### PR TITLE
fixed nav menu at the top on mobile and desktop

### PR DESCRIPTION
**Pull request by paulo-valim**
Screenshot 2023-02-07 at 14 45 59

Screenshot 2023-02-07 at 14 45 54

### CODE REVIEW

**Comment by josevalim** [comment]
Can you please add a small border (1px solid #EEE) to the bottom of the header? Like this:

image

However, the border should only appear once we start scrolling.

**Comment by jonatanklosko** [comment]
@josevalim I think the floating sidebar takes way too much space, perhaps less vertical padding?

**Comment by josevalim** [comment]
Sounds good. Once we start scrolling, we can reduce the padding too.

**Comment by paulo-valim** [comment]
Done! =)

**Comment by josevalim** [comment]
@jonatanklosko please merge if you are happy :) it LGTM!

File: src/js/main.js [hunk 1]
```diff
@@ -362,3 +362,31 @@ if (document.body.dataset.page === "integrations") {
...
+
+window.addEventListener("scroll", function () {
+  if (window.scrollY > 1) {
+    headerMobile.classList.remove("header-mobile-fixed");
+    headerMobile.classList.add("header-mobile-scrolled");
+  } else {
+    headerMobile.classList.remove("header-mobile-scrolled");
+    headerMobile.classList.add("header-mobile-fixed");
+  }
+});
```
Comment (jonatanklosko) [inline src/js/main.js:392]: We only need the information if the element should have the scrolled variant or not, so we can add a single class like this:

```suggestion
// Change header style

const header = document.querySelector("header");
const headerMobile = document.querySelector("#header-mobile");

window.addEventListener("scroll", () => {
  const scrolled = window.scrollY > 1;
  header.classList.toggle("scrolled", scrolled);
  headerMobile.classList.toggle("scrolled", scrolled);
});
```

then you can have `<header class="header">` and a CSS rules like

```css
.header {
  /* base style */
}

.header.scrolled {
  /* overrides */
}

.header-mobile {...}
.header-mobile.scrolled {...}
```

When the `.scrolled` class is added, the additional styles will apply :)

File: src/partials/layout.html [hunk 2]
```diff
@@ -43,53 +43,54 @@
...
-            <img src="/images/badge_logo.svg" alt="" width="17" height="19" />
-            <span class="mb-[-1.5px] font-semibold">Create badge</span>
-          </a>
-          <a
-            class="p-2 flex items-center justify-center rounded-lg bg-white border border-gray-300 text-gray-500 hover:bg-gray-100 focus:bg-gray-100 text-xl leading-none"
-            href="/settings.html"
-            aria-label="settings"
-          >
-            <i class="ri-settings-3-line"></i>
+    <header class="header-fixed">
```
Comment (jonatanklosko) [inline src/partials/layout.html:46]:
```suggestion
    <header id="header" class="header-fixed">
```

so we can consistently query for `#header` and `#header-mobile` from JS.

**Comment by jonatanklosko** [comment]
LGTM! Just one suggestion how to simplify the logic :)

**Comment by paulo-valim** [comment]
Done! =)

File: src/css/components.css [hunk 3]
```diff
@@ -1,4 +1,22 @@
 @layer components {
+  /* Headers */
+
+  .header {
+    @apply sticky top-0 z-50 px-4 py-6 hidden lg:flex bg-white;
+  }
+
+  .header.scrolled {
+    @apply sticky top-0 z-50 px-4 py-3 hidden lg:flex bg-white border-b border-gray-200;
```
Comment (jonatanklosko) [inline src/css/components.css:9]:
```suggestion
    @apply py-3border-b border-gray-200;
```

those already come from `.header`, same for mobile :)

**Comment by jonatanklosko** [comment]
> header.classList.toggle("scrolled", scrolled);
> 
> Just one question, why did you pass the second argument >scrolled<?

It adds or removes the class depending on whether `scrolled` is true or false. Basically a shorthand for:

```js
if (scrolled) {
  header.classList.add("scrolled");
} else {
  header.classList.remove("scrolled");
}
```

**Review by jonatanklosko — APPROVED** [review]
🐈

**Comment by paulo-valim** [comment]
Done!

I didn't erased cause I thought it was going to replace somehow the .header.

I started doing the header component because if I passed the class py-3, shows an error. 

When doing .class1.class2, why is there no problem with having the class twice?

**Comment by jonatanklosko** [comment]
> When doing .class1.class2, why is there no problem with having the class twice?

```css
.header {
  @apply py-6;
}

.header.scroll {
  @apply py-3;
}
```

the `@apply` basically inlines the styling rules from the tailwind classes, so this translates to:

```css
.header {
  padding-top: 24px;
  padding-bottom: 24px;
}

.header.scroll {
  padding-top: 12px;
  padding-bottom: 12px;
}
```

Then, if the element has `class="header scroll"`, padding from the second definition takes precedence because `.header.scroll` is a more specific selector. If the selectors are equally specific, like `.a { padding: 2px; }` and `.b { padding: 4px; }`, hen the one later in the file takes precedence :)

**Comment by jonatanklosko** [comment]
And if your question was what `.header.scroll` refers to, it's a selector for element that has both `header` class and `scroll` class!

**Pull request merged**